api/src/main/java/io/minio/MinioClient.java

```diff
@@ -119,6 +123,9 @@
  * }</pre>
  */
 public class MinioClient extends S3Base {
+	
+  private ExecutorService executor;
```

Comment:
My recommendation is to keep async as separate client i.e. `AsyncMinioClient`. I am not sure this is the one we wanted in async support. I did something similar long back https://github.com/balamurugana/minio-java/commit/d6f5878c7d4100eafb0d7ae59506e1f0e7b6e0b1 but we would need to have full support from bottom to top i.e. from `OkHttpClient` to `MinioClient`.